arrows/mvg/applets/track_features.cxx

```diff
@@ -305,25 +397,38 @@ run()
     std::string video_source = config->get_value<std::string>("video_source");
     std::string mask_list_file = config->get_value<std::string>("mask_list_file");
     bool invert_masks = config->get_value<bool>("invert_masks");
+    bool use_video_metadata = config->get_value<bool>("use_video_metadata");
     bool expect_multichannel_masks = config->get_value<bool>("expect_multichannel_masks");
     std::string output_tracks_file = config->get_value<std::string>("output_tracks_file");
-
+    unsigned int max_frames = config->get_value<unsigned int>("feature_tracker:max_frames",
+                                                              500);
 
     LOG_INFO( main_logger, "Reading Video" );
     video_reader->open(video_source);
+    kwiver::vital::metadata_map_sptr md_map = video_reader->metadata_map();
 
-    // Pre-scan the video to get an accurate frame count
-    // We may wish to remove this later if we start operating on live streams
-    kwiver::vital::timestamp ts;
-    std::vector<kwiver::vital::timestamp> timestamps;
-    while( video_reader->next_frame(ts) )
+    std::vector<kwiver::vital::frame_id_t> valid_frames;
+    if (use_video_metadata && md_map && md_map->size() > 0)
     {
-      timestamps.push_back(ts);
+      auto fs = md_map->frames();
+      valid_frames = std::vector<kwiver::vital::frame_id_t>(fs.begin(), fs.end());
     }
-    // close and re-open to return to the video start
-    video_reader->close();
-    video_reader->open(video_source);
+    else
+    {
+      auto const num_frames = static_cast<kwiver::vital::frame_id_t>(
+        video_reader->num_frames());
+      valid_frames.reserve(num_frames);
+      for (kwiver::vital::frame_id_t f = 1; f <= num_frames; ++f)
+      {
+        valid_frames.push_back(f);
+      }
+    }
+
+    std::vector<kwiver::vital::frame_id_t> camera_frames;
```

Comment:
I had forgotten that previously the `camera_frames` was just for restarting previous results that already had cameras.  More commonly, it was for cases where metadata was used to create initial guess for the cameras.  That is, this helps align which frames we track to which frames have metadata.

For now I suggested with initialize `camera_frames` with the frames of `md_map` if metadata is being used or empty if not.  That's not exactly what the GUI does.  The GUI converts metadata to cameras first.  It's possible that some frames with metadata will not have sufficient metadata to make a camera, so you will get fewer camera frames.  However, in most cases, `camera_frames` will be the same as the frames on which we have metadata.